arex-storage-web-api/src/main/java/com/arextest/storage/interceptor/RequestResponseMetricInterceptor.java

```diff
@@ -0,0 +1,62 @@
+package com.arextest.storage.interceptor;
+
+import static com.arextest.diff.utils.JacksonHelperUtil.objectMapper;
+import com.arextest.storage.metric.RequestResponseMetricService;
+import com.dianping.cat.utils.StringUtils;
+import com.fasterxml.jackson.databind.JsonNode;
+import javax.annotation.Resource;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.stereotype.Component;
+import org.springframework.web.servlet.HandlerInterceptor;
+import org.springframework.web.util.ContentCachingRequestWrapper;
+import org.springframework.web.util.ContentCachingResponseWrapper;
+
+/**
+ * record request and response interceptor created by xinyuan_wang on 2023/11/5
+ */
+@Component
+@Slf4j
+public class RequestResponseMetricInterceptor implements HandlerInterceptor {
+
+  @Resource
+  private RequestResponseMetricService requestResponseMetricService;
+
+  /**
+   * Cache the response and record the size of the request and response in metrics
+   */
+  @Override
+  public void afterCompletion(HttpServletRequest request, HttpServletResponse response,
+      Object handler, Exception ex) throws Exception {
+    String requestURI = request.getRequestURI();
+    long requestBodySize = request.getContentLengthLong();
+    String appId = null;
+    String recordId = null;
+    String category = null;
+    if (request instanceof ContentCachingRequestWrapper) {
+      byte[] contentAsByteArray = ((ContentCachingRequestWrapper) request).getContentAsByteArray();
+      String requestBody = new String(contentAsByteArray);
+      if (StringUtils.isNotEmpty(requestBody)) {
+        JsonNode jsonNode = objectMapper.readTree(requestBody);
```

Comment:
Parsing the request body is not recommended, metric tags can be transmitted via the request header